clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2190,17 +2190,17 @@ LogicalResult UnaryOp::verify() {
   case cir::UnaryOpKind::Dec: {
     // TODO: Consider looking at the memory interface instead of
     // LoadOp/StoreOp.
-    auto loadOp = getInput().getDefiningOp<cir::LoadOp>();
-    if (!loadOp)
-      return emitOpError() << "requires input to be defined by a memory load";
-
-    for (const auto user : getResult().getUsers()) {
-      auto storeOp = dyn_cast<cir::StoreOp>(user);
-      if (storeOp && storeOp.getAddr() == loadOp.getAddr())
-        return success();
+    if (auto loadOp = getInput().getDefiningOp<cir::LoadOp>()) {
```

Comment:
I believe the original intent here was to closely map inc/dec behavior in C/C++, however, I don't see how that helps with anything yet and it's been a while we introduce it. Also, it's easier for any pass that cares to inspect the source and trace memory usage.

That said, let's make this even more simple: please update `UnaryOp` and remove the part that mention such load/store expectation and also the verifier checks.

Reply:
@bcardosolopes  so.. I removed these load/store checks and updated docs as well